res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -1444,69 +1444,132 @@ let wheelTimerDelta = 0;
-    constructor(unitNumber, inReports, outReport, io) {
+    constructor(numbers, inReports, outReport, io) {
         super();
-        this.group = `[EffectRack1_EffectUnit${unitNumber}]`;
-        this.unitNumber = unitNumber;
+        if (!Array.isArray(numbers)) {
+            console.warn("ERROR! new S4Mk3EffectUnit() called without specifying any unit numbers!");
+            return;
+        }
+        if (numbers.length > 2) {
+            console.warn("ERROR! new S4Mk3EffectUnit() called with more than 2 unit numbers!");
+            return;
+        }
+        let numCount = 0;
+        for (const num of numbers) {
+            if (typeof num === "number") {
+                numCount++;
+            }
+        }
+        if (numCount < numbers.length) {
+            console.warn("ERROR! new S4Mk3EffectUnit() called with array that contains other types than numbers!");
+            return;
+        }
+        this.unitNumbers = numbers;
+
+        this.unitNumber = this.unitNumbers[0];
+        this.group = this.unitGroupForNumber(this.unitNumber);
         this.focusedEffect = null;
+        this.focusSelectMode = false;
+        // With this control skins can mark the unit that is currently mapped to the controller
+        engine.setValue(this.group, "controller_input_active", 1);
+        // Unset for the other unit
+        const otherUnitGroup = this.unitGroupForNumber(this.unitNumbers[1]);
+        engine.setValue(otherUnitGroup, "controller_input_active", 0);
+        // From now on we only toggle it in setCurrentUnit()
 
         this.mixKnob = new Pot({
             inKey: "mix",
             group: this.group,
             inReport: inReports[2],
             inByte: io.mixKnob.inByte,
+            // TODO Super knob when shifted?
         });
 
-        this.mainButton = new PowerWindowButton({
+        // Focus button
+        // shortpress: clear focused effect, return to Meta/toggle mode
+        // shortpress + Shift: switch unit
+        // longpress: focus select mode
+        // LED: blinking in focus mode, else off
+        this.effectFocusButton = new PowerWindowButton({
             unit: this,
+            group: this.group,
             inReport: inReports[1],
-            inByte: io.mainButton.inByte,
-            inBit: io.mainButton.inBit,
-            outByte: io.mainButton.outByte,
+            inByte: io.effectFocusButton.inByte,
+            inBit: io.effectFocusButton.inBit,
+            outByte: io.effectFocusButton.outByte,
             outReport: outReport,
-            shift: function() {
-                this.group = this.unit.group;
-                this.outKey = "group_[Master]_enable";
-                this.outConnect();
-                this.outTrigger();
-            },
-            unshift: function() {
-                this.outDisconnect();
-                this.outKey = undefined;
-                this.group = undefined;
-                this.output(false);
-            },
```

Comment:
This feature, which allows to enable post fader FX seem to be entirely lost. Could we move it so it can still be used somewhere? (e.g shift+long press)